packages/gitbook/src/components/hooks/useHash.tsx

```diff
@@ -0,0 +1,47 @@
+'use client';
+import React from 'react';
+
+export const HashContext = React.createContext<{
+    hash: string | null;
+    updateHashFromUrl: (href: string) => void;
+}>({
+    hash: null,
+    updateHashFromUrl: () => {},
+});
+
+function getHash(): string | null {
+    if (typeof window === 'undefined') {
+        return null;
+    }
+    return window.location.hash.slice(1);
+}
+
+export const HashProvider: React.FC<React.PropsWithChildren<{}>> = ({ children }) => {
+    const [hash, setHash] = React.useState<string | null>(getHash);
```

Comment:
```suggestion
    const [hash, setHash] = React.useState<string | null>(getHash());
```

not sure how typescript didn't catch this one 👀